arwen/elrondapi/managedei.go

```diff
@@ -0,0 +1,1035 @@
+		return nil, err
+	}
+
+	imports, err = imports.Append("managedDelegateExecution", v1_4_managedDelegateExecution, C.v1_4_managedDelegateExecution)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("managedExecuteReadOnly", v1_4_managedExecuteReadOnly, C.v1_4_managedExecuteReadOnly)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("managedCreateContract", v1_4_managedCreateContract, C.v1_4_managedCreateContract)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("managedDeployFromSourceContract", v1_4_managedDeployFromSourceContract, C.v1_4_managedDeployFromSourceContract)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("managedUpgradeContract", v1_4_managedUpgradeContract, C.v1_4_managedUpgradeContract)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("managedUpgradeFromSourceContract", v1_4_managedUpgradeFromSourceContract, C.v1_4_managedUpgradeFromSourceContract)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("managedAsyncCall", v1_4_managedAsyncCall, C.v1_4_managedAsyncCall)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("managedESDTTransfers", v1_4_managedESDTTransfers, C.v1_4_managedESDTTransfers)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("managedGetESDTBalance", v1_4_managedGetESDTBalance, C.v1_4_managedGetESDTBalance)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("managedGetESDTTokenData", v1_4_managedGetESDTTokenData, C.v1_4_managedGetESDTTokenData)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("managedGetReturnData", v1_4_managedGetReturnData, C.v1_4_managedGetReturnData)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("managedGetPrevBlockRandomSeed", v1_4_managedGetPrevBlockRandomSeed, C.v1_4_managedGetPrevBlockRandomSeed)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("managedGetBlockRandomSeed", v1_4_managedGetBlockRandomSeed, C.v1_4_managedGetBlockRandomSeed)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("managedGetStateRootHash", v1_4_managedGetStateRootHash, C.v1_4_managedGetStateRootHash)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("managedGetOriginalTxHash", v1_4_managedGetOriginalTxHash, C.v1_4_managedGetOriginalTxHash)
+	if err != nil {
+		return nil, err
+	}
+
+	return imports, nil
+}
+
+//export v1_4_managedSCAddress
+func v1_4_managedSCAddress(context unsafe.Pointer, destinationHandle int32) {
+	managedType := arwen.GetManagedTypesContext(context)
+	runtime := arwen.GetRuntimeContext(context)
+	metering := arwen.GetMeteringContext(context)
+
+	gasToUse := metering.GasSchedule().ElrondAPICost.GetSCAddress
+	metering.UseGas(gasToUse)
+
+	scAddress := runtime.GetSCAddress()
+
+	managedType.SetBytes(destinationHandle, scAddress)
```

Comment:
`scAddress` is a pointer owned by the `RuntimeContext`, then passed directly to `managedType.SetBytes()` to be stored under `destinationHandle`.

If a contract α calls a contract β using `managedExecuteOnDestContext` by creating a managed buffer `Addressβ` (owned by α), then β may freely modify the buffer containing its address and that change will persist when returning to α. If α wants to use the `Addressβ` buffer for any purposes, it has been modified and does not contain the address of β anymore.

These bytes must be copied here.